pkg/karmadactl/promote.go

```diff
@@ -493,3 +496,26 @@ func buildClusterPropagationPolicy(gvr schema.GroupVersionResource, opts Command
 	}
 	return cpp
 }
+
+// BuildPropagationPolicyName build both propagationPolicy and clusterPropagationPolicy name according to resource FederatedKey
+func BuildPropagationPolicyName(opts CommandPromoteOption, gvk schema.GroupVersionKind, clusterScope bool) string {
```

Comment:
We can define the name functions in `pkg/util/names/names.go` and try simplify parmaters.

like
https://github.com/karmada-io/karmada/blob/1df916c1387abd412a48a2c6b8fe54612e27650f/pkg/util/names/names.go#L74-L75

Reply:
A little confused :
We do not use pointer while pass opts so we can get gvk from a opts in other place
Do you think func GeneratePolicyName(name string, namespace string, gvk schema.GroupVersionKind) is ok
Or do you have any idea about how to simplify parmaters.

Reply:
dirctly using `gvk string`?  Like `GeneratePolicyName(name string, namespace string, gvk string) `, we seems not need to use `federatedKey`.

Reply:
@lonelyCZ please take a look at the diffs